components/feature/prompts/src/androidTest/java/mozilla/components/feature/prompts/file/OnDeviceFilePickerTest.kt

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.file
+
+import android.content.ClipData
+import android.content.Context
+import android.content.Intent
+import androidx.core.net.toUri
+import androidx.test.core.app.ApplicationProvider
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.prompt.PromptRequest
+import mozilla.components.feature.prompts.PromptContainer
+import org.junit.Assert
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class OnDeviceFilePickerTest {
+    private val context: Context
+        get() = ApplicationProvider.getApplicationContext()
+
+    private val badUri1 = "file:///data/user/0/${context.packageName}/any_directory/file.text".toUri()
+    private val badUri2 = "file:///data/data/${context.packageName}/any_directory/file.text".toUri()
+    private val goodUri = "file:///data/directory/${context.packageName}/any_directory/file.text".toUri()
+
+    @Test
+    fun filterUrisUnderPrivateAppDirectory() {
+        val uris = arrayOf(badUri1, badUri2, goodUri)
+
+        val filterUris = uris.filterUrisUnderPrivateAppDirectory(context)
+
+        Assert.assertEquals(1, filterUris.size)
```

Comment:
This fails for me on an Android 11 emulator.
Not sure why..
but the `data/user/0` symlink is not resolved to `data/data`.
Don't have a physical device with A11 to test..

Reply:
@kanytu  Do you have any thoughts on this? As the variations between Android 10 and 11 can affect the reliability of `isUnderPrivateAppDirectory` as it is.

Reply:
@csadilek any thoughts ? :)

Reply:
I can confirm that in Android R, symlinks are not getting resolved with canonical path. I'm unsure if thats intended (as part of Scoped Storage) or if that's a bug. 

Either way, using `dataDir` instead of an hardcoded `/data/data/` will always be sure to match against private files, so I'm pretty sure this is still a reliable approach.

Reply:
Actually, I was wrong. If that doesn't resolve `/data/data/`, `/data/data/` files will still be considered non-private files. I would say that this should be checked once a stable version of R lands.

Reply:
How about we add a check for `data/user` specifically (in addition to the canonical path comparison) so the tests pass and we cover this case for current Android 11, where the symlink isn't resolved. We can file a follow-up ticket to clean this up / check again once Android 11 is out. This seems acceptable as we don't support `file://` URIs anyway right now?

Reply:
it sounds reasonable to me!